superduperdb/components/schema.py

```diff
@@ -124,12 +127,20 @@ def decode_data(self, data: dict[str, t.Any]) -> dict[str, t.Any]:
 
             value = data[k]
 
-            if isinstance(value, str) and value.startswith('?'):
-                decoded[k] = field.encodable_cls.build_from_reference(
-                    value, datatype=field
-                )
+            if isinstance(value, str) and value.startswith('&'):
+                kwargs = {}
+
+                if value.startswith('&:blob:'):
+                    file_id = value.split('&:blob:')[1]
+                    kwargs['blob'] = blobs.get(file_id, value)
```

Comment:
If the data has not been inserted into the database and the referenced data is stored in blobs, check first if the data already exists; if not, retrieve it from the artifact store.